### PR TITLE
Added support for nodes with no properties

### PR DESCRIPTION
**Pull request by jimboca**
Ryan,

I was able to use the library with my ISY on 4.3.10 and python3.  It seems the minodom on python2 didn't like my xml.   But it threw an exception because some of my nodes don't have properties, like this Z-Wave node.

<node flag="0">
<address>ZW003_144</address>
<name>ZW 003 Pulse Meter</name>
<family>4</family>
<parent type="3">63777</parent>
<type>4.32.1.0</type>
<enabled>true</enabled>
<deviceClass>0</deviceClass>
<wattage>0</wattage>
<dcPeriod>0</dcPeriod>
<pnode>ZW003_1</pnode>
<sgid>144</sgid>
<devtype>
<gen>4.32.1</gen>
<mfg>134.2.5</mfg>
<cat>144</cat>
<model>9</model>
</devtype>
<ELK_ID>M06</ELK_ID>
</node>

I do have changes that change to using ElementTree for nodes instead of minidom, but ET only accepts files, so I had to write the data to a file and pass that to ET, which I wasn't sure was a good way to handle it.

Hopefully this all looks good, since like I mentioned, this is my first time with Python and git, after a lot of years Perl and cvs/svn.

- Jim

### CODE REVIEW

**Comment by rmkraus** [comment]
Thanks for the pull request. I'll try to review it in detail tomorrow. In the meantime, could you do me a favor and post an example of one of the Z-Wave nodes inside of triple back ticks? This will force GitHub to display the xml properly. For an example, click on the "Markdown supported" link above the comment box and check out a code example. Thanks again!

File: PyISY/Nodes/__init__.py [hunk 1]
```diff
@@ -178,12 +178,15 @@ def parse(self, xml):
...
-                            nval = feature.getElementsByTagName('property')[0] \
-                                .attributes['value'].value
-                            dimmable = '%' in \
-                                feature.getElementsByTagName('property')[0] \
-                                .attributes['uom'].value
-                            nval = int(nval.replace(' ', '0'))
+                            nval = None
+                            dimmable = False
+                            nprop = feature.getElementsByTagName('property')
+                            #print(nname,nid,nprop,len(nprop))
```
Comment (rmkraus) [inline PyISY/Nodes/__init__.py]: This line should probably be removed.

**Comment by rmkraus** [comment]
Actually, that was pretty easy to review. :smile: 

Looks pretty good to me, lets just remove that one line. Additionally, if you could bump the version number to 1.0.6 in [setup.py](https://github.com/automicus/PyISY/blob/master/setup.py#L5) as well as [the init file](https://github.com/automicus/PyISY/blob/master/PyISY/__init__.py#L26), that would be helpful.

This will get by with handling Z-Wave nodes, but will probably not provide full functionality. It is probably good enough for now though. In the next year or so, I'll need to rewrite the nodes module to allow for different control variables. This will become increasingly important when the ISY firmware 5.0 comes out.

**Comment by rmkraus** [comment]
Also, welcome to the wonderful world of Python and Git! The Python vs Perl argument is one that will probably rage for decades, but distributed version control is a lot more friendly than centralized version control once you get used to it. GitHub definitely helps with the learning curve.

**Comment by jimboca** [comment]
Great, glad you like it. I can send you examples of Z-Wave xml, and will make the other changes you requested tomorrow. Some Z-Wave devices are controllable and have the property, just not all. 

Yes, the 5.0 firmware is my main motivation for getting familiar with all this now so I'll be ready to work with the nodeserver function when it gets a little more mature. 

I'm liking Python so far, just a lot to learn. Git is way better and we have been trying to move to it for a while, but big companies can move slow.

**Comment by jimboca** [comment]
Requested changes made.

Here is some xml examples for this device which is the older Aeon multi-sensor, with a couple of the infamous phantom nodes for Pulse Meter and Window Alarm...

```xml
<node flag="128">
<address>ZW003_1</address>
<name>Back Deck MS</name>
<family>4</family>
<parent type="3">63777</parent>
<type>4.32.1.0</type>
<enabled>true</enabled>
<deviceClass>0</deviceClass>
<wattage>0</wattage>
<dcPeriod>0</dcPeriod>
<pnode>ZW003_1</pnode>
<sgid>1</sgid>
<devtype>
<gen>4.32.1</gen>
<mfg>134.2.5</mfg>
<cat>104</cat>
<model>9</model>
</devtype>
<ELK_ID>K04</ELK_ID>
<property id="ST" value="0" formatted="Off" uom="78"/>
</node>
<node flag="0">
<address>ZW003_118</address>
<name>Back Deck</name>
<family>4</family>
<parent type="3">63777</parent>
<type>4.32.1.0</type>
<enabled>true</enabled>
<deviceClass>0</deviceClass>
<wattage>0</wattage>
<dcPeriod>0</dcPeriod>
<pnode>ZW003_1</pnode>
<sgid>118</sgid>
<devtype>
<gen>4.32.1</gen>
<mfg>134.2.5</mfg>
<cat>118</cat>
<model>9</model>
</devtype>
<ELK_ID>K05</ELK_ID>
</node>
<node flag="0">
<address>ZW003_143</address>
<name>ZW 003 Energy Meter</name>
<family>4</family>
<parent type="3">63777</parent>
<type>4.32.1.0</type>
<enabled>true</enabled>
<deviceClass>0</deviceClass>
<wattage>0</wattage>
<dcPeriod>0</dcPeriod>
<pnode>ZW003_1</pnode>
<sgid>143</sgid>
<devtype>
<gen>4.32.1</gen>
<mfg>134.2.5</mfg>
<cat>143</cat>
<model>9</model>
</devtype>
<property id="ST" value="6800" formatted="68.00 Watts" uom="73" prec="2"/>
</node>
<node flag="0">
<address>ZW003_144</address>
<name>ZW 003 Pulse Meter</name>
<family>4</family>
<parent type="3">63777</parent>
<type>4.32.1.0</type>
<enabled>true</enabled>
<deviceClass>0</deviceClass>
<wattage>0</wattage>
<dcPeriod>0</dcPeriod>
<pnode>ZW003_1</pnode>
<sgid>144</sgid>
<devtype>
<gen>4.32.1</gen>
<mfg>134.2.5</mfg>
<cat>144</cat>
<model>9</model>
</devtype>
<ELK_ID>M06</ELK_ID>
</node>
<node flag="0">
<address>ZW003_160</address>
<name>ZW 003 Window Alarm</name>
<family>4</family>
<type>4.32.1.0</type>
<enabled>true</enabled>
<deviceClass>0</deviceClass>
<wattage>0</wattage>
<dcPeriod>0</dcPeriod>
<pnode>ZW003_1</pnode>
<sgid>160</sgid>
<devtype>
<gen>4.32.1</gen>
<mfg>134.2.5</mfg>
<cat>160</cat>
<model>9</model>
</devtype>
<property id="ST" value="0" formatted="Off" uom="78"/>
</node>
```

Hopefully the pull request is correct.

**Comment by rmkraus** [comment]
Looks good to me. I'm not sure I want to add the example scripts into the main repository though. Would you mind removing them until I merge the pull request?

**Comment by jimboca** [comment]
Sorry, I was thinking the pull request was tied to a specific version of the branch, not the latest version. The are deleted. They were just for my reference for now and I'll put them in another repo if necessary, that is if I can get them to even run on my RPi...

**Comment by rmkraus** [comment]
Thanks, looks good. As for getting the tests to run on your RPi, make sure to install the apscheduler module:

sudo pip install apscheduler

Also, be sure to close the yaml file when you are done with it (stream.close()) just because it is the right thing to do. Other than that, it looks like it should work, but I'm not familiar with apscheduler so I really have no idea. :wink:

**Comment by jimboca** [comment]
Thanks for taking my changes. I can't even get the yaml module to work on my rpi and python3, it worked great on a redhat machine. Need to do more research.

**Pull request merged**